internal/validation/apiextensions/v1/xrd/handler.go

```diff
@@ -145,16 +146,18 @@ func (v *validator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.O
 }
 
 func (v *validator) dryRunUpdateOrCreateIfNotFound(ctx context.Context, crd *apiextv1.CustomResourceDefinition) error {
-	got := crd.DeepCopy()
-	err := v.client.Get(ctx, client.ObjectKey{Name: crd.Name}, got)
-	if err == nil {
-		got.Spec = crd.Spec
-		return v.client.Update(ctx, got, client.DryRunAll)
-	}
-	if kerrors.IsNotFound(err) {
-		return v.client.Create(ctx, crd, client.DryRunAll)
-	}
-	return err
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		got := crd.DeepCopy()
+		err := v.client.Get(ctx, client.ObjectKey{Name: crd.Name}, got)
+		if err == nil {
+			got.Spec = crd.Spec
```

Comment:
preexisting: doesn't this mean that `Update` below can mutate `crd`? The json unmarshaller under the hood might reuse referenced data structures.

Reply:
The semantics of this function are fishy. `crd` might get some changes on update, but not all.

Reply:
out of scope of the PR, just saying.

Reply:
I believe the deep copy is preserving the original object. We pass a pointer to the deep copied object in the `Update` call.

Reply:
The object `got` is fresh, but `crd.Spec` is not. The DeepCopy does not help for that.

Reply:
IIRC that's the point, I was getting some error due to something else being updated, so I only switched in the part I care validating, the spec. Happy to drop it if that's not needed.